pkg/resourcemanager/controller/managedresource/reconciler.go

```diff
@@ -173,6 +179,8 @@ func (r *reconciler) reconcile(ctx context.Context, mr *resourcesv1alpha1.Manage
 			return reconcile.Result{}, fmt.Errorf("could not read secret '%s': %+v", secret.Name, err)
 		}
 
+		secretChecksum = secretChecksum + utils.ComputeChecksum(secret.Data)
```

Comment:
Can you calculate a single checksum of all secrets instead of appending for each secret? This is actually how checksums are designed to be used and also how it is implemented in go. Please see https://pkg.go.dev/crypto/sha256 for examples.

Reply:
> Can you calculate a single checksum of all secrets instead of appending for each secret? This is actually how checksums are designed to be used and also how it is implemented in go. Please see https://pkg.go.dev/crypto/sha256 for examples.

The checksum is calculated for each referenced secret's data. If a Managed resource is having more than one secret present in `.spec.secretRefs`, a checksum needs to be calculated for each secret's data. How can we calculate the checksum of all referenced secrets at once? In any case, we need to traverse through each referenced secret and append the calculated checksum.

Reply:
Now I get it, you mean instead of calculating the checksum for each secret data and then appending it, we can append the hash of each secret and then calculate the checksum at last. Hope I got it right this time?

Reply:
Made the changes in [commit](https://github.com/gardener/gardener/pull/6136/commits/47235b9c43d325412e674b0d15b6de2c56f089db).